modules/evm-bridge/src/lib.rs

```diff
@@ -72,6 +74,89 @@ pub mod module {
 }
 
 impl<T: Config> EVMBridgeTrait<AccountIdOf<T>, BalanceOf<T>> for Pallet<T> {
+	// Calls the name method on an ERC20 contract using the given context
+	// and returns the token name.
+	fn name(context: InvokeContext) -> Result<Vec<u8>, DispatchError> {
+		// ERC20.name method hash
+		let input = hex!("06fdde03").to_vec();
+
+		let info = T::EVM::execute(context, input, Default::default(), 2_100_000, 0, ExecutionMode::View)?;
+
+		Self::handle_exit_reason(info.exit_reason)?;
+
+		// output is 32-byte aligned and consists of 3 parts:
+		// - part 1: 32 byte, the offset of its description is passed in the position of
+		// the corresponding parameter or return value.
+		// - part 2: 32 byte, string length
+		// - part 3: string data
+		let output = info.output.as_slice().to_vec();
+		ensure!(
+			output.len() >= 64 && output.len() % 32 == 0,
+			Error::<T>::InvalidReturnValue
+		);
+
+		let offset = U256::from_big_endian(&output[0..32]);
+		let length = U256::from_big_endian(&output[offset.as_usize()..offset.as_usize() + 32]);
+		ensure!(
+			output.len() >= offset.as_usize() + 32 + length.as_usize(),
+			Error::<T>::InvalidReturnValue
+		);
+
+		let mut data = Vec::new();
+		data.extend_from_slice(&output[offset.as_usize() + 32..offset.as_usize() + 32 + length.as_usize()]);
+
+		Ok(data.to_vec())
+	}
+
+	// Calls the symbol method on an ERC20 contract using the given context
+	// and returns the token symbol.
+	fn symbol(context: InvokeContext) -> Result<Vec<u8>, DispatchError> {
+		// ERC20.symbol method hash
+		let input = hex!("95d89b41").to_vec();
+
+		let info = T::EVM::execute(context, input, Default::default(), 2_100_000, 0, ExecutionMode::View)?;
+
+		Self::handle_exit_reason(info.exit_reason)?;
+
+		// output is 32-byte aligned and consists of 3 parts:
+		// - part 1: 32 byte, the offset of its description is passed in the position of
+		// the corresponding parameter or return value.
+		// - part 2: 32 byte, string length
+		// - part 3: string data
+		let output = info.output.as_slice().to_vec();
+		ensure!(
+			output.len() >= 64 && output.len() % 32 == 0,
+			Error::<T>::InvalidReturnValue
+		);
+
+		let offset = U256::from_big_endian(&output[0..32]);
+		let length = U256::from_big_endian(&output[offset.as_usize()..offset.as_usize() + 32]);
+		ensure!(
+			output.len() >= offset.as_usize() + 32 + length.as_usize(),
+			Error::<T>::InvalidReturnValue
+		);
+
+		let mut data = Vec::new();
+		data.extend_from_slice(&output[offset.as_usize() + 32..offset.as_usize() + 32 + length.as_usize()]);
```

Comment:
This part the exactly the same with function `name`. Making output to data conversion as a function would make it easier to read and avoid repeats.